test/test_quantumchannel.jl

```diff
@@ -0,0 +1,109 @@
+using QuantumSavory
+using ResumableFunctions
+using ConcurrentSim
+using Test
+
+bell = (Z1⊗Z1 + Z2⊗Z2)/sqrt(2.0)
+regA = Register(2)
+regB = Register(2)
+initialize!((regA[1], regB[2]), bell)
+
+sim = Simulation()
+
+# Delay queue for quantum channel 
+queue = DelayQueue{Register}(sim, 10.0)
+
+qc = QuantumChannel(queue)
+
+@resumable function alice_node(env, qc)
+    put!(qc, regA[1])
+end
+
+@resumable function bob_node(env, qc)
+    @yield @process take!(env, qc, regB[1])
```

Comment:
Why is the `@process` necessary here?

Reply:
If I remove the `@ process` from here, with everything else remaining the same, it errors with:
```julia
ERROR: MethodError: no method matching state(::var"##Base.take!_FSMI#315"{QuantumChannel, Any, RegRef, Any})

Closest candidates are:
  state(::AbstractEvent)
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\base.jl:45

Stacktrace:
 [1] execute(ev::ConcurrentSim.Initialize, proc::Process)
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\processes.jl:37
 [2] (::ConcurrentSim.var"#1#2"{typeof(ConcurrentSim.execute), ConcurrentSim.Initialize, Tuple{Process}})()
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\base.jl:51
 [3] step(sim::Simulation)
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\simulations.jl:39
 [4] run(env::Simulation, until::ConcurrentSim.Timeout)
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\base.jl:91
 [5] run
   @ C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\events.jl:29 [inlined]
 [6] run(env::Simulation)
   @ ConcurrentSim C:\Users\Abhishek Bhatt\.julia\dev\ConcurrentSim\src\events.jl:29
 [7] top-level scope
   @ c:\Users\Abhishek Bhatt\.julia\dev\channels.jl:70
```

Reply:
and if I remove the `@yield` as well, doesn't error but skips any computation after the `@yield` in `take!`, not sure how to get around this behaviour

Reply:
It seems like the @process is necessary here because without the `QuantumSavory.take!` being a process, the yield in `QuantumSavory.take!` is not a part of the env and hence it doesn't wait on the `get` object returned by https://github.com/JuliaDynamics/ConcurrentSim.jl/blob/8591bff85edb62556be142f141dafa506e0f7fae/src/resources/delayed_stores.jl#L57

The take! method is kinda similar to the consumer process in this example: https://github.com/JuliaDynamics/ConcurrentSim.jl/blob/8591bff85edb62556be142f141dafa506e0f7fae/src/resources/delayed_stores.jl#L16-L22

Reply:
The issue is that we want to keep this `take!` consistent with the one for the other types of stores. In all the other cases `take!` only needs a `@yield`, presumably because it already returns a process type.